faker/providers/internet/__init__.py

```diff
@@ -169,20 +199,50 @@ def url(self):
         pattern = self.random_element(self.url_formats)
         return self.generator.parse(pattern)
 
-    def ipv4(self, network=False):
-        """Produce a random IPv4 address or network with a valid CIDR"""
-        address = str(ip_address(self.generator.random.randint(
-            0, (2 ** IPV4LENGTH) - 1)))
-        if network:
+    def ipv4(self, network=False, private=False):
+        """Produce a random IPv4 address or network with a valid CIDR."""
+        if private is True:
+            networks = dict()
+            for i in range(len(_private_ipv4_networks)):
+                networks[i] = _private_ipv4_networks[i].num_addresses
+            k = self.generator.random_element(networks)
+            _net = _private_ipv4_networks[k]
```

Comment:
I'm not sure I'm following this logic.  I don't see the point of the `for` loop. Seems to me that it's picking up a random item of the `_private_ipv4_networks` list to assign to `_net` and could be done with just `_net = self.generator.random.choice(_private_ipv4_networks)`

Reply:
would you put this explanation in a comment?